src/main/java/edu/hw3/Task2.java

```
@@ -0,0 +1,39 @@
+package edu.hw3;
+
+import java.util.ArrayList;
+import java.util.Stack;
+
+public class Task2 {
+    private Task2() {
+
+    }
+
+    public static ArrayList<String> clusterize(String str) {
+        if (str == null) {
+            return null;
+        }
+        if (str.charAt(0) == ')') {
+            return null;
```

Лучше кидать исключение в таких случаях. Иначе можно получить отложенный NPE в дальнейшем флоу. Принцип fail fast.